src/swarm/neo/node/ConnectionHandler.d

```diff
@@ -357,6 +375,43 @@ class ConnectionHandler : IConnectionHandler
 
     private FinalizeDg return_to_pool;
 
+    /***************************************************************************
+
+        Type of a delegate that provides a scope-allocated request resource
+        acquirer, via the provided delegate, for use in a request handler.
+
+        Params:
+            shared_resources = opaque shared resources instance owned by the
+                node. Should be cast to the node-specific shared resources
+                object
+            handle_request_dg = delegate that receives a resources acquirer and
+                initiates handling of a request
+
+    ***************************************************************************/
+
+    public alias void delegate ( Object shared_resources,
+        void delegate ( Object resource_acquirer ) handle_request_dg ) AcquirerDg;
+
+    /***************************************************************************
+
+        Delegate that provides a scope-allocated request resource acquirer for
+        use in request handlers.
+
+    ***************************************************************************/
+
+    private AcquirerDg acquirer_dg;
+
+    /***************************************************************************
+
+        Re-usable buffer used to emplace request handler objects into. (Instead
+        of heap allocating an instance every time a request is handled.) We use
+        a heap buffer, rather than a fixed-size array on the stack, in order to
+        not have to guess the maximum size of a request handler object.
+
+    ***************************************************************************/
+
+    private void[] emplace_buf;
```

Comment:
Shouldn't this be on the `RequestOnConn` level, given that the single connection handles many requests?

Reply:
Haha, probably yes.

Reply:
That'd be a big advantage of stack allocation, actually.

Reply:
Could we plug in swarms `AcquiredBuffers` here? So that the `handleRequest` gets a buffer and recycles it?

Reply:
Hmm... In theory, yes. The code in swarm doesn't currently know anything about any kind of shared resources, though.

Reply:
No, I mean just to have reusable buffers, nothing else.

Reply:
What would be the advantage?

Reply:
I wrote that comment before I've heard it's easy enough to add it as a member of the `RequestOnConn` :-)

Reply:
Updated.